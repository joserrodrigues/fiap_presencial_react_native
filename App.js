import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={styles.textInfo}>Hello World</Text>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    fontSize: 40,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInfo: {
    fontSize: 35,
    fontWeight: 'bold',
    color: 'red'
  }
});
