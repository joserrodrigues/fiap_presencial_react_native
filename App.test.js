import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import App from './App';

describe('<App />', () => {
    it('Check Initial Case', async () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree.children.length).toBe(1);
        expect(tree).toMatchSnapshot();
    });

    it('Check Button Press', async () => {
        const page = render(<App />);
        const textInfo = page.getByText("Teste Rubens");
        expect(textInfo).toBeTruthy();

        const button = page.getByTestId('buttonID');
        expect(button).toBeTruthy();
        fireEvent.press(button);

        const successMessage = await waitFor(() =>
            page.getByText("Teste 2")
        );
        expect(successMessage).toBeTruthy();
    });
});