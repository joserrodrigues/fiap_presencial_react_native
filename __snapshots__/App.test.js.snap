// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Check Initial Case 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 30,
      }
    }
  >
    <Text>
      Teste Rubens
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="buttonID"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Teste
        </Text>
      </View>
    </View>
  </View>
</View>
`;
